modules/puppetca_token_whitelisting/puppetca_token_whitelisting_autosigner.rb

```diff
@@ -0,0 +1,96 @@
+require 'jwt'
+require 'openssl'
+
+module ::Proxy::PuppetCa::TokenWhitelisting
+  class Autosigner
+    include ::Proxy::Log
+    include ::Proxy::Util
+
+    def sign_all
+      Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.sign_all
+    end
+
+    def hosts_file
+      Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.hosts_file
+    end
+
+    def smartproxy_cert
+      OpenSSL::PKey::RSA.new File.read Proxy::SETTINGS.ssl_private_key.to_s
+    end
+
+    def ensure_hostsfile
+      unless File.exist?(hosts_file)
+        FileUtils.mkdir_p hosts_file.rpartition('/').first
+        FileUtils.touch hosts_file
+        File.write(hosts_file, [].to_yaml)
```

Comment:
I'd move writing to this file to a (private) helper function rather than duplicate it in 3 places.